game-core/src/main/java/games/strategy/triplea/ai/tree/BattleStep.java

```diff
@@ -0,0 +1,1216 @@
+        hasResult = true;
+      } else {
+        // Get all allied transports in the territory
+        final Predicate<Unit> matchAllied =
+            Matches.unitIsTransport()
+                .and(Matches.unitIsNotCombatTransport())
+                .and(Matches.isUnitAllied(attacker, parameters.getData()));
+        final List<Unit> alliedTransports =
+            CollectionUtils.getMatches(battleSite.getUnits(), matchAllied);
+        // If no transports, just end the battle
+        if (alliedTransports.isEmpty()) {
+          loseProbability = 1;
+          hasResult = true;
+        } else if (round <= 1) {
+          /*
+          TODO: update the attacking units in units
+          attackingUnits =
+              CollectionUtils.getMatches(
+                  battleSite.getUnits(), Matches.unitIsOwnedBy(attacker));
+           */
+        } else {
+          loseProbability = 1;
+          hasResult = true;
+        }
+      }
+    } else if (CollectionUtils.getMatches(defendingUnits, Matches.unitIsNotInfrastructure()).size()
+        == 0) {
+      if (Properties.getTransportCasualtiesRestricted(parameters.getData())) {
+        // If there are undefended attacking transports, determine if they automatically die
+        checkUndefendedTransports(units.getEnemy(), units);
+      }
+      checkForUnitsThatCanRollLeft(false, units);
+      winProbability = 1;
+      hasResult = true;
+    } else if (MAX_ROUNDS > 0 && MAX_ROUNDS <= round) {
+      tieProbability = 1;
+      hasResult = true;
+      /*
+         * Removed the else because this is slow for every iteration and
+         * is already taken care of by getFightOutcomes
+      } else {
+        final Collection<TerritoryEffect> territoryEffects = parameters.getTerritoryEffects();
+        final int attackPower =
+            DiceRoll.getTotalPower(
+                DiceRoll.getUnitPowerAndRollsForNormalBattles(
+                    attackingUnits,
+                    defendingUnits,
+                    attackingUnits,
+                    false,
+                    parameters.getData(),
+                    battleSite,
+                    territoryEffects,
+                    parameters.isAmphibious,
+                    parameters.amphibiousLandAttackers),
+                parameters.getData());
+        final int defensePower =
+            DiceRoll.getTotalPower(
+                DiceRoll.getUnitPowerAndRollsForNormalBattles(
+                    defendingUnits,
+                    attackingUnits,
+                    defendingUnits,
+                    true,
+                    parameters.getData(),
+                    battleSite,
+                    territoryEffects,
+                    parameters.isAmphibious,
+                    parameters.amphibiousLandAttackers),
+                parameters.getData());
+        if (attackPower == 0 && defensePower == 0) {
+          tieProbability = 1;
+          hasResult = true;
+        }
+         */
```

Comment:
Commented out code should be remove in the end

Reply:
Agreed.  But this is in code copied from MustFightBattle and I wanted to indicate why that else statement was commented out.  Once the duplicate code from MustFightBattle has been put in a common place, this will no longer be needed.